scripts/dex-router/Dockerfile

```diff
@@ -56,10 +46,7 @@ ENV NODE_ENV production
 # Run the application as a non-root user.
 USER node
 
-# Copy package.json so that package manager commands can be used.
-COPY package.json .
-
 # Copy the production dependencies from the deps stage and also
 # the built application from the build stage into the image.
-COPY --from=deps /usr/src/app/node_modules ./node_modules
-COPY --from=build /usr/src/app/dist ./dist
+COPY --from=build /prod/dex-router/node_modules ./node_modules
+COPY --from=build /usr/src/app/scripts/dex-router/dist ./dist
```

Comment:
I did a few things in `Dockerfile.node` to optimize stuff (https://github.com/pyth-network/pyth-crosschain/blob/main/Dockerfile.node) that you might want to duplicate:

1. Base image shas are locked, preventing issues where some tag floated and you upgrade dependencies unintentionally (these can be very painful to debug)
2. The build image and the runner images are separated; a large image is used for the build which prevents some issues with some packages that Alpine has with building some gyp-based dependencies, these can again be very painful to debug
3. Corepack is used to manage the pnpm version, so you don't need to worry about changing the PNPM_VERSION to match what you're using in dev

Honestly you can probably just use the exact same dockerfile if you wanted

Reply:
As I understand, I think the difference here is that you are copying the whole monorepo to build packages. I'm not sure if I wanna do same thing for building the packages :-?

Reply:
ahh yeah that part isn't well optimized, I was more focused on optimizing the output image size and avoiding build issues from Node dependencies that can be painful to deal with than I was focused on improving the build time.  I think you could probably combine some of the optimizations above with also only copying specific packages into the build image...